packages/core/src/routes/sign-in-experience.ts

```diff
@@ -41,11 +72,39 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
       body: SignInExperiences.createGuard.omit({ id: true }).partial(),
     }),
     async (ctx, next) => {
-      const { body } = ctx.guard;
+      const { socialSignInConnectorIds, ...rest } = ctx.guard.body;
+      const { signInMethods } = rest;
+
+      if (signInMethods) {
+        const connectorInstances = await getConnectorInstances();
+
+        if (isEnabled(signInMethods.email)) {
+          assertEnabledConnectorByType(ConnectorType.Email, connectorInstances);
+        }
+
+        if (isEnabled(signInMethods.sms)) {
+          assertEnabledConnectorByType(ConnectorType.SMS, connectorInstances);
+        }
+
+        if (isEnabled(signInMethods.social)) {
+          assertEnabledConnectorByType(ConnectorType.Social, connectorInstances);
+
+          assertThat(
+            socialSignInConnectorIds && socialSignInConnectorIds.length > 0,
+            'sign_in_experiences.empty_social_connectors'
+          );
+
+          assertEnabledSocialConnectorsByIds(socialSignInConnectorIds, connectorInstances);
+        }
+      }
+
+      // Only when the parameter signInMethods.social is enabled, do update socialSignInConnectorIds.
```

Comment:
nit: Only update socialSignInConnectorIds when the parameter signInMethods.social is enabled

Reply:
Or `Only update socialSignInConnectorIds when social sign-in is enabled`?

Updated, PTAL.